JIT IR - Make valueMapPtr optional in convertNetDefToIR

duc0 opened the pull request:
Summary: Make valueMapPtr optional in convertNetDefToIR, and add tests

Differential Revision: D14429687

ZolotukhinM commented on torch/csrc/jit/netdef_converter.h:
```diff
@@ -11,7 +11,7 @@ namespace jit {
  * torch::jit::Graph \p graph. The function also records name->value map in \p
  * valueMapPtr. If the original net had several values with the same name, the
- * map will contain the value for the last definition.
+ * map will contain the value for the last definition. valueMapPtr is optional.
```
Maybe we should then add a `nullptr` default value for `valueMapPtr` argument then?

ZolotukhinM reviewed (approved):
Looks good! Please find a minor remark inline and don't forget to rebase as the first commit has landed already.

duc0 committed:
JIT IR - Make valueMapPtr optional in convertNetDefToIR (#17942)

Summary:
Make valueMapPtr optional in convertNetDefToIR, and add tests
Pull Request resolved: https://github.com/pytorch/pytorch/pull/17942

Differential Revision: D14429687

fbshipit-source-id: 496d37607dbe4272a7ea3eed7578b48b64b4fc4c
